enderio-machines/src/main/java/com/enderio/machines/common/blocks/wireless_charger/WirelessAntennaBlock.java

```diff
@@ -0,0 +1,43 @@
+package com.enderio.machines.common.blocks.wireless_charger;
+
+import net.minecraft.core.Direction;
+import net.minecraft.world.item.context.BlockPlaceContext;
+import net.minecraft.world.level.block.Block;
+import net.minecraft.world.level.block.state.BlockState;
+import net.minecraft.world.level.block.state.StateDefinition;
+import net.minecraft.world.level.block.state.properties.BlockStateProperties;
+import net.minecraft.world.level.block.state.properties.DirectionProperty;
+import net.minecraft.world.level.block.state.properties.IntegerProperty;
+import net.neoforged.neoforge.common.ModConfigSpec;
+import org.jetbrains.annotations.Nullable;
+
+public class WirelessAntennaBlock extends Block {
+
+    private static final DirectionProperty FACING = BlockStateProperties.HORIZONTAL_FACING;
+
+    public static final IntegerProperty WIRELESS_RANGE = IntegerProperty.create("wireless_charger_range", 0, 2048);
```

Comment:
Using a blockstate property for this is interesting. I just wonder if components + attachments might be better?

Reply:
On blocks we should store data in the block entity. I mistakenly adopted attachments when we ported to Neo and have been working to undo this work.

Attachments - for adding data to stuff that isn't "yours".
Once MC changes to components however this story changes :)